providers/catalog.go

```diff
@@ -673,30 +673,31 @@ var catalog = CatalogType{ // nolint:gochecknoglobals
 	},
 
 	// Docusign configuration
-	// TODO: we don't have a good way to get the server string yet. Need to make API call to /oauth/userinfo.
-	// Leaving this connector commented out until that is unblocked.
-	// Docusign: {
-	//	AuthType: Oauth2,
-	//	BaseURL:  "https://{{.server}}.docusign.net",
-	//	OauthOpts: OauthOpts{
-	//		AuthURL:                   "https://account.docusign.com/oauth/auth",
-	//		TokenURL:                  "https://account.docusign.com/oauth/token",
-	//		ExplicitScopesRequired:    true,
-	//		ExplicitWorkspaceRequired: false,
-	//	},
-	//	Support: Support{
-	//		BulkWrite: BulkWriteSupport{
-	//			Insert: false,
-	//			Update: false,
-	//			Upsert: false,
-	//			Delete: false,
-	//		},
-	//		Proxy:     false,
-	//		Read:      false,
-	//		Subscribe: false,
-	//		Write:     false,
-	//	},
-	// },
+	Docusign: {
+		AuthType: Oauth2,
+		BaseURL:  "https://{{.server}}.docusign.net",
+		OauthOpts: OauthOpts{
+			AuthURL:                   "https://account.docusign.com/oauth/auth",
+			TokenURL:                  "https://account.docusign.com/oauth/token",
+			ExplicitScopesRequired:    true,
+			ExplicitWorkspaceRequired: false,
+		},
+		Support: Support{
+			BulkWrite: BulkWriteSupport{
+				Insert: false,
+				Update: false,
+				Upsert: false,
+				Delete: false,
+			},
+			Proxy:     false,
+			Read:      false,
+			Subscribe: false,
+			Write:     false,
+		},
+		ProviderOpts: ProviderOpts{
+			"userInfoURL": "https://account.docusign.com/oauth/userinfo",
```

Comment:
Instead of this being in `ProviderOpts`, can we make this a top-level field to track all URLs that we need to call in an additional step after oauth flow (e.g. Hubspot's [GetAccountDetails](https://legacydocs.hubspot.com/docs/methods/get-account-details), Mailchimp's OAuth metadata endpoint). I'm imagining something like this:

```
Docusign: {
		AuthType: Oauth2,
		BaseURL:  "https://{{.server}}.docusign.net",
                UserInfoURL: "https://account.docusign.com/oauth/userinfo",
```

Alternative names for this field:
- AccountInfoURL
- ConnectionInfoURL
- ConnectionMetadataURL

Reply:
What do you think of storing a reference to the method instead? So it could look like - 
```
         Docusign: {
		AuthType: Oauth2,
		BaseURL:  "https://{{.server}}.docusign.net",
                FetchConnectionMetadata: docusign.GetUserInfo,
```

Reply:
I'm not a big fan of this approach since it's optimizing for having different method names for different connectors. (i.e. `docusign.GetUserInfo`, `mailchimp.GetPrefix`, etc.), which I don't think is needed. For consistency, I'd much rather always use the same method name for all connectors' "post-auth metadata collection" step, just like how all connectors that support read have a `Read` function. The method name can be something more generic like `getPostAuthInfo` or `getConnectionMetadata`. 

Then the catalog can either have:
- a boolean value like `postAuthInfoNeeded`, which tells the server whether or not to invoke the `getPostAuthInfo` method
- a string value like `postAuthInfoURL` or `userInfoURL`, and the presence of the string value tells the server whether or not to invoke the `getPostAuthInfo` method.

Reply:
Going with `postAuthInfoNeeded`, which will assume that connectors will implement `GetPostAuthInfo`